common/dynamicconfig/constants.go

```diff
@@ -3456,6 +3463,11 @@ var IntKeys = map[IntKey]DynamicInt{
 		Description:  "ReplicationTaskProcessorErrorRetryMaxAttempts is the max retry attempts for applying replication tasks",
 		DefaultValue: 10,
 	},
+	MaxWorkflowCacheSize: DynamicInt{
+		KeyName:      "history.MaxWorkflowCacheSize",
+		Description:  "is the max number of workflows for which we store in memory information",
+		DefaultValue: 1000,
```

Comment:
A few clarification questions:
1. where is this going to be used exactly and is it unlimited right now? 
2. Is this going to be per-domain or per-shard limit? 
3. what is the potential side effects of limiting to 1k?